**step-1: import packages and read the data**

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


In [ ]:
step2: seperate 

**step:3 Frequency**

In [ ]:
# number of applicants from 'Asia'
visa_df #df
visa_df['continent'] # one column
con=visa_df['continent']=='Asia'  #one class
np.sum(con) # 1+1+1+1+0+1+0+1
visa_df[con] # extraction

**Value_Counts**

In [1]:
visa_df['continent'].value_counts()

NameError: name 'visa_df' is not defined

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


In [2]:
visa_df=pd.read_csv(r"C:\Users\lavan\Downloads\Visadataset.csv")
visa_df

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
1,EZYV02,Asia,Master's,Y,N,2412,2002,Northeast,83425.6500,Year,Y,Certified
2,EZYV03,Asia,Bachelor's,N,Y,44444,2008,West,122996.8600,Year,Y,Denied
3,EZYV04,Asia,Bachelor's,N,N,98,1897,West,83434.0300,Year,Y,Denied
4,EZYV05,Africa,Master's,Y,N,1082,2005,South,149907.3900,Year,Y,Certified
...,...,...,...,...,...,...,...,...,...,...,...,...
25475,EZYV25476,Asia,Bachelor's,Y,Y,2601,2008,South,77092.5700,Year,Y,Certified
25476,EZYV25477,Asia,High School,Y,N,3274,2006,Northeast,279174.7900,Year,Y,Certified
25477,EZYV25478,Asia,Master's,Y,N,1121,1910,South,146298.8500,Year,N,Certified
25478,EZYV25479,Asia,Master's,Y,Y,1918,1887,West,86154.7700,Year,Y,Certified


In [3]:
visa_df['continent']

0          Asia
1          Asia
2          Asia
3          Asia
4        Africa
          ...  
25475      Asia
25476      Asia
25477      Asia
25478      Asia
25479      Asia
Name: continent, Length: 25480, dtype: object

In [4]:
visa_df['continent'].unique()

array(['Asia', 'Africa', 'North America', 'Europe', 'South America',
       'Oceania'], dtype=object)

In [5]:
visa_df['continent'].nunique()

6

- there are 25480 applicants from 6 continents
- number of applicants from each continent:Frequency table

In [8]:
visa_df['continent'].value_counts()

continent
Asia             16861
Europe            3732
North America     3292
South America      852
Africa             551
Oceania            192
Name: count, dtype: int64

In [16]:
keys=visa_df['continent'].value_counts().keys().tolist()
values=visa_df['continent'].value_counts().values.tolist()

In [17]:
keys

['Asia', 'Europe', 'North America', 'South America', 'Africa', 'Oceania']

In [18]:
values

[16861, 3732, 3292, 852, 551, 192]

In [20]:
visa_df=pd.DataFrame(zip(keys,values),columns=['class','frequency'])
visa_df
                     
                     
                         

,class,frequency
0,Asia,16861
1,Europe,3732
2,North America,3292
3,South America,852
4,Africa,551
5,Oceania,192


In [23]:
visa_df.groupby(by='continent').count()['case_id']

KeyError: 'continent'